### PR TITLE
[Reviewer: Adam]Add a config plugin for the BGCF json file.

### PR DESCRIPTION
**Pull request by kenferguson**
Can you review please? I'll be committing this then deleting the version that's merged to CC since we don't need it.

### CODE REVIEW

File: clearwater_config_access/bgcf_json_config_plugin.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+class BgcfJson(ConfigType):
+    name = 'bgcf_json'
+    filetype = 'json'
+    # file_download_name is used to agree with the current naming system when
+    # writing to file.
+    file_download_name = 'bgcf.json'
+
+    # This help_info appears as user-visible help text in the usage statement
+    # for cw-config.
+    help_info = ('''bgcf_json - this stores the configuration of each
```
Comment (johadalin) [inline clearwater_config_access/bgcf_json_config_plugin.py]: Given the next line, any reason this is split across 2 lines? Feels like it'd be fine if it didn't have the triple quotes, but wa s on a single line

File: clearwater_config_access/bgcf_json_config_plugin.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+                    BGCF.''')
+    schema = '/usr/share/clearwater/clearwater-config-manager/scripts/config_validation/bgcf_schema.json'
```
Comment (johadalin) [inline clearwater_config_access/bgcf_json_config_plugin.py]: Is this going to exists in PC? I'm not sure who creates the schemas, and whether this is not the Project Clearwater BGCF one

**Comment by johadalin** [comment]
Seems sensible. You've got a `pragma no-cover` on the load_as_plugin method. Does that mean the rest of the file is properly tested? :D 
We should probably patch this on to a system to check it works, but if it does, all seems good

File: clearwater_config_access/bgcf_json_config_plugin.py (hunk 2 above)
Comment (kenferguson) [reply clearwater_config_access/bgcf_json_config_plugin.py]: I believe it's present on sprout nodes (it's in the sprout repository under the sprout-bgcf.root directory). I'll live test to make sure though.

File: clearwater_config_access/bgcf_json_config_plugin.py (hunk 1 above)
Comment (kenferguson) [reply clearwater_config_access/bgcf_json_config_plugin.py]: No - will change.

**Pull request merged**